diff --git a/node_modules/@shopware-ag/meteor-component-library/dist/mt-datepicker.vue_vue_type_style_index_0_lang-3628874c.mjs b/node_modules/@shopware-ag/meteor-component-library/dist/mt-datepicker.vue_vue_type_style_index_0_lang-3628874c.mjs
index e318098..f45da1f 100644
--- a/node_modules/@shopware-ag/meteor-component-library/dist/mt-datepicker.vue_vue_type_style_index_0_lang-3628874c.mjs
+++ b/node_modules/@shopware-ag/meteor-component-library/dist/mt-datepicker.vue_vue_type_style_index_0_lang-3628874c.mjs
@@ -11043,11 +11043,13 @@ const _sfc_main = /* @__PURE__ */ defineComponent({
     let localeLoadId = 0;
     async function importDateFnsLocaleModule(path) {
       try {
-        const { default: locale } = await import(
+        const localeModule = await import(
           /* @vite-ignore */
           `date-fns/locale/${path}`
         );
-        return locale;
+        const localeName = path.replace(/-/g, "");
+        const locale = localeModule[localeName] ?? localeModule.default?.[localeName] ?? localeModule.default?.default ?? localeModule.default ?? localeModule;
+        return locale?.localize ? locale : null;
       } catch {
         return null;
       }
diff --git a/node_modules/@shopware-ag/meteor-component-library/dist/mt-datepicker.vue_vue_type_style_index_0_lang-9222785c.js b/node_modules/@shopware-ag/meteor-component-library/dist/mt-datepicker.vue_vue_type_style_index_0_lang-9222785c.js
index 9bf4ab6..98a5ad2 100644
--- a/node_modules/@shopware-ag/meteor-component-library/dist/mt-datepicker.vue_vue_type_style_index_0_lang-9222785c.js
+++ b/node_modules/@shopware-ag/meteor-component-library/dist/mt-datepicker.vue_vue_type_style_index_0_lang-9222785c.js
@@ -11044,11 +11044,13 @@ const _sfc_main = /* @__PURE__ */ vue.defineComponent({
     let localeLoadId = 0;
     async function importDateFnsLocaleModule(path) {
       try {
-        const { default: locale } = await import(
+        const localeModule = await import(
           /* @vite-ignore */
           `date-fns/locale/${path}`
         );
-        return locale;
+        const localeName = path.replace(/-/g, "");
+        const locale = localeModule[localeName] ?? localeModule.default?.[localeName] ?? localeModule.default?.default ?? localeModule.default ?? localeModule;
+        return locale?.localize ? locale : null;
       } catch {
         return null;
       }
