diff --git a/node_modules/eslint-plugin-vue/lib/utils/indent-common.js b/node_modules/eslint-plugin-vue/lib/utils/indent-common.js
index d2879b1..0b0fba8 100644
--- a/node_modules/eslint-plugin-vue/lib/utils/indent-common.js
+++ b/node_modules/eslint-plugin-vue/lib/utils/indent-common.js
@@ -198,7 +198,10 @@ module.exports.defineVisitor = function create(
   tokenStore,
   defaultOptions
 ) {
-  if (!context.getFilename().endsWith('.vue')) return {}
+  if (
+      !context.getFilename().endsWith('.vue') &&
+      !context.getFilename().endsWith('.twig')
+  ) return {}
 
   const options = parseOptions(
     context.options[0],
