diff --git a/bin/roave-backward-compatibility-check.php b/bin/roave-backward-compatibility-check.php
index 8909d19..0b48abc 100644
--- a/bin/roave-backward-compatibility-check.php
+++ b/bin/roave-backward-compatibility-check.php
@@ -28,6 +28,8 @@ use Roave\BackwardCompatibility\LocateDependencies\LocateDependenciesViaComposer
 use Roave\BackwardCompatibility\LocateSources\LocateSourcesViaComposerJson;
 use Roave\BetterReflection\BetterReflection;
 use RuntimeException;
+use Shopware\RoaveBackwardCompatibility\Check\ClassBased\AnnotationClassCheck;
+use Shopware\RoaveBackwardCompatibility\Check\MethodBased\AnnotationMethodCheck;
 use Symfony\Component\Console\Application;
 use Symfony\Component\Console\Input\ArgvInput;
 use Symfony\Component\Console\Output\ConsoleOutput;
@@ -83,6 +85,7 @@ use function file_exists;
         new CompareClasses(
             new ClassBased\SkipClassBasedErrors(new ClassBased\ExcludeAnonymousClasses(new ClassBased\ExcludeInternalClass(
                 new ClassBased\MultipleChecksOnAClass(
+                    new ClassBased\SkipClassBasedErrors(new AnnotationClassCheck()),
                     new ClassBased\SkipClassBasedErrors(new ClassBased\ClassBecameAbstract()),
                     new ClassBased\SkipClassBasedErrors(new ClassBased\ClassBecameInterface()),
                     new ClassBased\SkipClassBasedErrors(new ClassBased\ClassBecameTrait()),
@@ -136,6 +139,7 @@ use function file_exists;
                                 new MethodBased\SkipMethodBasedErrors(new MethodBased\ExcludeInternalMethod(new MethodBased\MultipleChecksOnAMethod(
                                     new MethodBased\SkipMethodBasedErrors(new MethodBased\OnlyPublicMethodChanged(
                                         new MethodBased\MultipleChecksOnAMethod(
+                                            new MethodBased\SkipMethodBasedErrors(new AnnotationMethodCheck()),
                                             new MethodBased\SkipMethodBasedErrors(new MethodBased\MethodBecameFinal()),
                                             new MethodBased\SkipMethodBasedErrors(new MethodBased\MethodConcretenessChanged()),
                                             new MethodBased\SkipMethodBasedErrors(new MethodBased\MethodScopeChanged()),
