diff --git a/src/LocateDependencies/LocateDependenciesViaComposer.php b/src/LocateDependencies/LocateDependenciesViaComposer.php
index a5a1ce0..e2bcf48 100644
--- a/src/LocateDependencies/LocateDependenciesViaComposer.php
+++ b/src/LocateDependencies/LocateDependenciesViaComposer.php
@@ -40,19 +40,8 @@ final class LocateDependenciesViaComposer implements LocateDependencies
     {
         Assert::file($installationPath . '/composer.json');
 
-        $this->runInDirectory(function () use ($installationPath): void {
-            $installer = ($this->makeComposerInstaller)($installationPath);
-
-            assert($installer instanceof Installer);
-
-            // Some defaults needed for this specific implementation:
-            $installer->setDevMode(false);
-            $installer->setDumpAutoloader(false);
-            $installer->setRunScripts(false);
-            $installer->setIgnorePlatformRequirements(true);
-
-            $installer->run();
-        }, $installationPath);
+        exec(\sprintf('composer install -d %s --no-dev --no-scripts --ignore-platform-reqs --no-interaction', escapeshellarg($installationPath)));
+        $this->fixeMarc1706FastImageSizeDirectories($installationPath);
 
         return new AggregateSourceLocator([
             (new MakeLocatorForInstalledJson())->__invoke($installationPath, $this->astLocator),
@@ -71,4 +60,9 @@ final class LocateDependenciesViaComposer implements LocateDependencies
             chdir($originalDirectory);
         }
     }
+
+    private function fixeMarc1706FastImageSizeDirectories(string $installationPath)
+    {
+        mkdir($installationPath . '/vendor/marc1706/fast-image-size/tests');
+    }
 }
